src/modules/silentpayments/main_impl.h

```diff
@@ -116,7 +119,11 @@ int secp256k1_silentpayments_create_shared_secret(const secp256k1_context *ctx,
     }
 
     /* Compute shared_secret = tweaked_secret_component * Public_component */
-    if (!secp256k1_ecdh(ctx, shared_secret33, public_component, tweaked_secret_component, secp256k1_silentpayments_ecdh_return_pubkey, NULL)) {
+    if (!secp256k1_ec_pubkey_tweak_mul(ctx, &ecdhSecretPubKey, tweaked_secret_component)) {
```

Comment:
This is because `secp256k1_ecdh` is constant time, whereas `ec_pubkey_tweak_mul` is not. Its something we've been discussing, whether its safe / worth it to use `ec_mult` instead of forcing it to be constant time, but its nice to see some actual numbers! I'm a bit surprised the speedup is only 7%ish; an earlier comment indicated it might be up to 25% improvement (per https://gist.github.com/RubenSomsen/c43b79517e7cb701ebf77eec6dbb46b8?permalink_comment_id=4125475#gistcomment-4125475)

Reply:
Ahh yeah that makes sense, I wasn't sure about that. With regards to the speed up, the number in the title is wrong it's more ~9%. I'm also not sure where exactly the 25% come from but I do a general benchmark on creating pub keys not just computing the secret. It's possible that when we run a benchmark exclusively on the shared secret, that we'll see a bigger percentage difference between implementations.

Reply:
I ran similar benchmarks isolating the ecdh computation. I find a ~19% time reduction when using simple multiplication instead of constant time ecdh.